hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomStaticCallOperation.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.operations;
+
+import com.hedera.node.app.service.contract.impl.exec.AddressChecks;
+import com.hedera.node.app.service.contract.impl.exec.processors.CustomMessageCallProcessor;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.operation.StaticCallOperation;
+
+/**
+ * Hedera customization of {@link StaticCallOperation} that immediately halts on calls to missing addresses,
+ * <i>unless</i> the call is to an address in the system account range, in which case the fate of the call
+ * is determined by the {@link CustomMessageCallProcessor}.
+ */
+public class CustomStaticCallOperation extends StaticCallOperation implements BasicCustomCallOperation {
```

Comment:
(I'm kind of wondering now if there could be a **generic** `SpecificCustomCallOperation` - would we be able to write this boilerplate only once?)  (Like `SpecificCustomCallOperation<TCallOperation> extends TCallOperation implements BasicCustomCallOperation`?)

Reply:
That would be much nicer, but I wanted to use `StaticCallOperation.to()` and it's `protected` so requires a subclass. 😞

Reply:
eek - let me think on this